### PR TITLE
phaser3 rewrite

### PR DESCRIPTION
**Pull request by sdairs**
The version of Phaser is quite old, so I started migrating it to Phaser3. It's pretty much all working, but still need some polish.

Also brings in Phaser via npm & uses vite

### CODE REVIEW

**Comment by sdairs** [comment]
Added:

- Tinybird data project files with a top 10 sessions API
- Basic JS fetch to send individual point scores to Tinybird Events API

File: src/MainMenuScene.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      Phaser.Geom.Rectangle.Contains
+    );
+
+    this.add.text(170, 277.5, "Play", {
+      fontSize: "24px",
+      color: "#ffffff",
+    });
+
+    graphics.on(
+      "pointerup",
```
Comment (JoeKarlsson) [inline src/MainMenuScene.js:21]: We should add both keyboard press up, and touch input anywhere on the screen to start the next round. Feels clunky to have to switch to the mouse to start the next round.

File: src/FlappyTinybirdScene.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  }
+
+  create() {
+    this.bird = this.physics.add.sprite(100, 245, "bird");
+
+    const spaceKey = this.input.keyboard.addKey(
+      Phaser.Input.Keyboard.KeyCodes.SPACE
+    );
+
+    spaceKey.on("down", (key, event) => {
```
Comment (JoeKarlsson) [inline src/FlappyTinybirdScene.js:34]: We should make this mobile friendly. I added `this.input.onDown.add(this.jump, this);` in my old version to get this to work on mobile.

**Review by JoeKarlsson — APPROVED** [review]
This rewrite looks amazing. Nice work. Love how you were able to make the game easier to play ;) I have some minor changes that would be good to fix, but we can totally handle those in future PRs since this one is starting to get pretty big. 

Also, looks like Phaser debugger mode is still on too ;)

File: src/MainMenuScene.js (hunk 1 above)
Comment (sdairs) [reply src/MainMenuScene.js:30]: 👍 added

File: src/FlappyTinybirdScene.js (hunk 2 above)
Comment (sdairs) [reply src/FlappyTinybirdScene.js:43]: 👍 added

**Comment by sdairs** [comment]
1. Added touch input as well as space to jump.
2. Added a HTML form to input a name & email address for the scoreboard. You cannot press play without entering details.
3. Changed it so that the app container is sized & centered, rather than the canvas, because the DOM overlay is placed inside the same container at the same level as the canvas, so it was off-center

TODO:

Need to style the HTML form!
An 'end game' screen would be good, that shows your score, and gives you the option to 'Play again' without having to re-enter your name/email

**Comment by sdairs** [comment]
Top 10 Scoreboard API + read only auth token

https://api.tinybird.co/v0/pipes/top_10_sessions.json?token=p.eyJ1IjogIjYwOTQyZDQ5LTU1NzYtNDIyMC04YTRjLTZhOTVlMmU1N2RlNCIsICJpZCI6ICI1ODRmMWJhZi1lZTc4LTQ5MmUtOWIxYS1iZjcyNjk4MGFkZjAifQ.xZJUmNrN0f9MFO91-i-eA9_iJH25nFYeNruTn55SqQI

**Pull request merged**